cmd/prometheus/main.go

```diff
@@ -671,6 +674,7 @@ func main() {
 	}
 	if cfgFile.StorageConfig.TSDBConfig != nil {
 		cfg.tsdb.OutOfOrderTimeWindow = cfgFile.StorageConfig.TSDBConfig.OutOfOrderTimeWindow
+		cfg.tsdb.BlockReloadInterval = cfgFile.StorageConfig.TSDBConfig.BlockReloadInterval
```

Comment:
The flag value should only be overridden if `BlockReloadInterval` is actually set in the config.
```suggestion
		if cfgFile.StorageConfig.TSDBConfig.BlockReloadInterval > 0 {
			cfg.tsdb.BlockReloadInterval = cfgFile.StorageConfig.TSDBConfig.BlockReloadInterval
		}
```